### PR TITLE
CI: Build, test, deploy to maven

### LINKED ISSUE #3: Add CI

### PR DESCRIPTION
**Pull request by jaeseung-bae**
## Description

* Close #3 
* Add CI/CD
  * Build, test, deploy to maven

## Checklist:

- [x] I have added a relevant changelog to `CHANGELOG.md`.
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/workflows/tag.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Java
+        uses: actions/setup-java@v3
+        with:
+          java-version: 8
+          distribution: temurin
```
[the remark this answers is not in the document]
Comment (jaeseung-bae) [reply .github/workflows/tag.yml:16]: temurin is one of JDK distribution from eclipse.
JDK is necessary for building kotlin.

File: crypto/build.gradle.kts [hunk 2]
```diff
@@ -34,5 +35,84 @@ dependencies {
...
+val pomName = "finschia"
+val pomDesc = artifactIdVal
+val pomUrl = "https://github.com/Finschia/finschia-kt"
+val pomScmConnection = "scm:git:git://github.com/Finschia/finschia-kt.git"
+val pomDeveloperConnection = "scm:git:ssh://github.com/Finschia/finschia-kt.git"
+val pomScmUrl = "https://github.com/Finschia/finschia-kt"
+
+// Maven account
+val ossrhUserName = System.getenv("OSSRH_USERNAME")
+val ossrhPassword = System.getenv("OSSRH_PW")
```
Comment (zemyblue) [inline crypto/build.gradle.kts:66]: This configuration is the information to need in only our side. The other user to download this source does not need it. 
Is it no problem? And Isn't there any other way to use ci?

**Review by zemyblue — COMMENTED** [review]
Please change the title. 
And please add prefix `ci` into the title in this case.

File: crypto/build.gradle.kts [hunk 3]
```diff
@@ -37,7 +37,7 @@ dependencies {
     testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:$jupiterVersion")
 }
 
-java{
+java {
```
Comment (loin3) [inline crypto/build.gradle.kts:40]: this change does not applied in tx/build.gradle.kts.

Comment (jaeseung-bae) [reply crypto/build.gradle.kts:40]: I moved maven-related definition into publishing blocks in both tx/ and crypto/. 
Do you mean this??

Comment (loin3) [reply crypto/build.gradle.kts:40]: I mean line 18 in tx/buid.gradle.kts, there is no blank between `java{`.

Comment (jaeseung-bae) [reply crypto/build.gradle.kts:40]: Ok~

**Review by loin3 — APPROVED** [review]

File: .github/workflows/tag.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+name: Publish package to Maven Central Repository
+on:
+  push:
+    # Sequence of patterns matched against refs/tags
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+" # Push events to matching v*, i.e. v1.0, v20.15.10
```
Comment (loin3) [inline .github/workflows/tag.yml:6]: Are we not going to publish when rc tag is pushed?
like v1.0.0-rc2?

Comment (jaeseung-bae) [reply .github/workflows/tag.yml:6]: Oh, Good feedback, I'll add regex for rc*

**Review by zemyblue — DISMISSED** [review]

**Review by loin3 — APPROVED** [review]

**Review by zemyblue — APPROVED** [review]

**Pull request merged**